### PR TITLE
CoreNEURON CI via Github Actions

### PR DESCRIPTION
**Pull request by alexsavulescu**
CI_BRANCHES:NEURON_BRANCH=master

### CODE REVIEW

File: .github/workflows/coreneuron-ci.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        shell: bash
+        working-directory: ${{runner.workspace}}/CoreNeuron
+        run:  |
+          if [ -n "$GCC_VERSION" ]; then
+              export CXX="g++-${GCC_VERSION}" CC="gcc-${GCC_VERSION}";
+          fi
+          
+          if [[ "${{ startsWith(matrix.os, 'macOS') }}" = "true" ]]; then
+              git submodule update --init -- external/Random123
+              export PATH=${{runner.workspace}}/CoreNeuron/external/Random123:/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH;
```
Comment (pramodk) [inline .github/workflows/coreneuron-ci.yml:110]: Curious why random123 is added to PATH?

**Review by pramodk — APPROVED** [review]

File: .github/workflows/coreneuron-ci.yml (hunk 1 above)
Comment (alexsavulescu) [reply .github/workflows/coreneuron-ci.yml:110]: Some strange Random123 is found in the image. Had to put that first

Comment (pramodk) [reply .github/workflows/coreneuron-ci.yml:110]: Ok..but PATH is for binaries/executables.Random123 is header only library. I wonder how PATH affects that. Do you have build where it was failing?

Comment (alexsavulescu) [reply .github/workflows/coreneuron-ci.yml:110]: https://github.com/alexsavulescu/CoreNeuron/runs/1464884089?check_suite_focus=true

Comment (alexsavulescu) [reply .github/workflows/coreneuron-ci.yml:110]: CMake was finding this : `Found Random123: /Users/runner/hostedtoolcache/stack/2.5.1/x64`, and found that to be in the PATH.

Comment (pramodk) [reply .github/workflows/coreneuron-ci.yml:110]: Quite surprising !

**Pull request merged**